pkgs/desktops/cinnamon/cinnamon-screensaver/default.nix

```diff
@@ -82,9 +84,7 @@ stdenv.mkDerivation rec {
   NIX_CFLAGS_COMPILE = "-I${glib.dev}/include/gio-unix-2.0"; # TODO: https://github.com/NixOS/nixpkgs/issues/36468
 
   postPatch = ''
-    patchShebangs autogen.sh
-
-    sed ${stdenv.lib.escapeShellArg "s&DBUS_SESSION_SERVICE_DIR=.*&DBUS_SESSION_SERVICE_DIR=`$PKG_CONFIG --variable session_bus_services_dir dbus-1 | sed -e 's,/usr/share,\${datarootdir},g' | sed 's|^|$out|'`&g"} -i configure.ac
+    sed ${stdenv.lib.escapeShellArg "s|dbus_services_dir = .*|dbus_services_dir = 'share/dbus-1/services'|g"} -i meson.build
```

Comment:
this `sed` is obsolete since the definition in the meson.build is supremely compatible  with nixos https://github.com/linuxmint/cinnamon-screensaver/blob/master/meson.build#L46

Reply:
```
[0/1] Installing files.
Installing subdir /build/source/data/icons/hicolor to /nix/store/jbpibqbf8d9mzxp2fgv6pqgks0i26zvp-cinnamon-screensaver-4.6.0/share/icons/hicolor
Installing /build/source/data/icons/hicolor/scalable/actions/screensaver-unlock-symbolic.svg to /nix/store/jbpibqbf8d9mzxp2fgv6pqgks0i26zvp-cinnamon-screensaver-4.6.0/share/icons/hicolor/scalable/actions
Installing /build/source/data/icons/hicolor/scalable/actions/screensaver-switch-users-symbolic.svg to /nix/store/jbpibqbf8d9mzxp2fgv6pqgks0i26zvp-cinnamon-screensaver-4.6.0/share/icons/hicolor/scalable/actions
Installing /build/source/data/icons/hicolor/scalable/status/screensaver-blank.svg to /nix/store/jbpibqbf8d9mzxp2fgv6pqgks0i26zvp-cinnamon-screensaver-4.6.0/share/icons/hicolor/scalable/status
Installing /build/source/data/icons/hicolor/scalable/status/screensaver-notification-symbolic.svg to /nix/store/jbpibqbf8d9mzxp2fgv6pqgks0i26zvp-cinnamon-screensaver-4.6.0/share/icons/hicolor/scalable/status
Installing libcscreensaver/libcscreensaver.so.0.0.0 to /nix/store/jbpibqbf8d9mzxp2fgv6pqgks0i26zvp-cinnamon-screensaver-4.6.0/lib
Installing libcscreensaver/CScreensaver-1.0.gir to /nix/store/jbpibqbf8d9mzxp2fgv6pqgks0i26zvp-cinnamon-screensaver-4.6.0/share/gir-1.0
Installing libcscreensaver/CScreensaver-1.0.typelib to /nix/store/jbpibqbf8d9mzxp2fgv6pqgks0i26zvp-cinnamon-screensaver-4.6.0/lib/girepository-1.0
Installing src/pamhelper/cinnamon-screensaver-pam-helper to /nix/store/jbpibqbf8d9mzxp2fgv6pqgks0i26zvp-cinnamon-screensaver-4.6.0/libexec
Installing /build/source/libcscreensaver/cs-auth.h to /nix/store/jbpibqbf8d9mzxp2fgv6pqgks0i26zvp-cinnamon-screensaver-4.6.0/include/cinnamon-screensaver/libcscreensaver
Installing /build/source/libcscreensaver/setuid.h to /nix/store/jbpibqbf8d9mzxp2fgv6pqgks0i26zvp-cinnamon-screensaver-4.6.0/include/cinnamon-screensaver/libcscreensaver
Installing /build/source/build/meson-private/cscreensaver.pc to /nix/store/jbpibqbf8d9mzxp2fgv6pqgks0i26zvp-cinnamon-screensaver-4.6.0/lib/pkgconfig
Installing /build/source/data/cinnamon-screensaver to /nix/store/jbpibqbf8d9mzxp2fgv6pqgks0i26zvp-cinnamon-screensaver-4.6.0/etc/pam.d
Installing /build/source/build/data/cinnamon-screensaver.desktop to /nix/store/jbpibqbf8d9mzxp2fgv6pqgks0i26zvp-cinnamon-screensaver-4.6.0/share/applications
Traceback (most recent call last):
  File "/nix/store/cg1lyk9naby9pl4srl3igi5cca6qjzqc-meson-0.55.1/lib/python3.8/site-packages/mesonbuild/mesonmain.py", line 131, in run
    return options.run_func(options)
  File "/nix/store/cg1lyk9naby9pl4srl3igi5cca6qjzqc-meson-0.55.1/lib/python3.8/site-packages/mesonbuild/minstall.py", line 542, in run
    installer.do_install(datafilename)
  File "/nix/store/cg1lyk9naby9pl4srl3igi5cca6qjzqc-meson-0.55.1/lib/python3.8/site-packages/mesonbuild/minstall.py", line 365, in do_install
    self.install_data(d)
  File "/nix/store/cg1lyk9naby9pl4srl3igi5cca6qjzqc-meson-0.55.1/lib/python3.8/site-packages/mesonbuild/minstall.py", line 396, in install_data
    if self.do_copyfile(fullfilename, outfilename, makedirs=(d.dirmaker, outdir)):
  File "/nix/store/cg1lyk9naby9pl4srl3igi5cca6qjzqc-meson-0.55.1/lib/python3.8/site-packages/mesonbuild/minstall.py", line 263, in do_copyfile
    dirmaker.makedirs(outdir, exist_ok=True)
  File "/nix/store/cg1lyk9naby9pl4srl3igi5cca6qjzqc-meson-0.55.1/lib/python3.8/site-packages/mesonbuild/minstall.py", line 60, in makedirs
    os.makedirs(path, exist_ok=exist_ok)
  File "/nix/store/685kq8pyhrvajah1hdsfn4q7gm3j4yd4-python3-3.8.5/lib/python3.8/os.py", line 213, in makedirs
    makedirs(head, exist_ok=exist_ok)
  File "/nix/store/685kq8pyhrvajah1hdsfn4q7gm3j4yd4-python3-3.8.5/lib/python3.8/os.py", line 223, in makedirs
    mkdir(name, mode)
PermissionError: [Errno 13] Permission denied: '/etc/dbus-1'
FAILED: meson-install
```

Reply:
this is because they're doing `define_variable` on `prefix` when it should be `datadir`

Reply:
so we need a different sed then?

Reply:
here's the patch https://github.com/linuxmint/cinnamon-screensaver/pull/349/files